hpOneView/resources/networking/logical_switch_groups.py

```diff
@@ -0,0 +1,149 @@
+                if returning the requested number of items would take too long.
+            filter:
+                A general filter/query string to narrow the list of items returned. The
+                default is no filter - all resources are returned.
+            sort:
+                The sort order of the returned data set. By default, the sort order is based
+                on create time, with the oldest entry first.
+
+        Returns:
+            list: A list of  logical switch groups
+
+        """
+        return self._client.get_all(start, count, filter=filter, sort=sort)
+
+    def get(self, id_or_uri):
+        """
+        Gets a logical switch group by ID or by uri
+        Args:
+            id_or_uri: Could be either the logical switch group id or the logical switch group uri
+
+        Returns:
+            dict: The logical switch group
+        """
+        return self._client.get(id_or_uri)
+
+    def create(self, resource, timeout=-1):
+        """
+        Creates a logical switch group.
+        Args:
+            resource (dict): Object to create
+            timeout:
+                Timeout in seconds. Wait task completion by default. The timeout does not abort the operation
+                in OneView, just stops waiting for its completion.
+
+        Returns:
+            dict: Created logical switch group.
+
+        """
+        return self._client.create(resource, timeout=timeout)
+
+    def update(self, resource, timeout=-1):
+        """
+        Updates a logical switch group.
+        Args:
+            resource (dict): Object to update
+            timeout:
+                Timeout in seconds. Wait task completion by default. The timeout does not abort the operation
+                in OneView, just stops waiting for its completion.
+
+        Returns:
+            dict: Updated logical switch group.
+
+        """
+        return self._client.update(resource, timeout=timeout)
+
+    def delete(self, resource, force=False, timeout=-1):
+        """
+        Deletes a logical switch group.
+
+        Args:
+            resource (dict): object to delete
+            force (bool):
+                 If set to true the operation completes despite any problems with
+                 network connectivity or errors on the resource itself. The default is false.
+            timeout:
+                Timeout in seconds. Wait task completion by default. The timeout does not abort the operation
+                in OneView, just stops waiting for its completion.
+
+        Returns:
+            dict: Details of associated resource
+
+        """
+        return self._client.delete(resource, force=force, timeout=timeout)
+
+    def get_by(self, field, value):
+        """
+        Get all Logical switch groups that matches the filter
+        The search is case insensitive
+
+        Args:
+            field: field name to filter
+            value: value to filter
+
+        Returns: dict
```

Comment:
I think that if it gets all Logical switch groups that matches the filter, then it doesn't return dict, but a list